### DISCUSSION TITLE
Make alias descriptions E2EE

### DISCUSSION THREAD

**Discussion by ghost** [simple-login/app#1851] [Feature Requests]
So that we can save private information there for better alias management.

[Related post](https://www.reddit.com/r/Simplelogin/comments/163wndg/is_it_safe_to_include_sensitive_information_in/)

**Comment by LavaBit5** [comment]
Yeah. That would be great!

User can store their sensitive data (such as PINs or passwords).

So, @nguyenkims, we DEFINITELY need this feature. You are a part of Proton now so I hope it won’t be a big problem.

**Comment by Tapi894** [comment]
@nguyenkims and @acasajus is that planned?

**Comment by nguyenkims** [comment]
Currently it isn't high in our todo list and from what I see, requires SL to support:
- you upload a public key that we use to encrypt the alias note
- you set the private key somewhere so the alias note can be decrypted on SL page. 

The 2nd step can only be done when we change SL from a "classic" web app to a single page application (SPA). This is actually a quite big project as SL wasn't designed to become SPA. However this SPA rewrite is already in progress and we hope to be able to finish it this year. It'll also allow to normalize the way mobile & web clients interact with SL server.

**Reply by D1stract1on** [reply]
What the difference between “classic” and SPA?

And why not to use the same key that used by Proton if integration enabled (Link SL to Proton enabled)? Or it is a bad idea?

**Reply by nguyenkims** [reply]
In a "classic" web app, all data is sent from the server and the frontend doesn't have its own data. 
In SPA app, the frontend asks for data when it needs them and it can store its own data which the server doesn't know.

**Reply by nguyenkims** [reply]
>And why not to use the same key that used by Proton if integration enabled (Link SL to Proton enabled)? Or it is a bad idea?

SL doesn't have access to this key and as I understand, this key is only known when you log in on Proton but not when you link these 2 accounts.
